cli/src/semdep/parsers/poetry.py

```diff
@@ -8,29 +8,49 @@
 from typing import List
 from typing import Optional
 
+from parsy import any_char
 from parsy import string
 from parsy import success
 
 from semdep.parsers.util import mark_line
-from semdep.parsers.util import not_any
 from semdep.parsers.util import safe_path_parse
 from semdep.parsers.util import transitivity
+from semdep.parsers.util import upto
 from semgrep.semgrep_interfaces.semgrep_output_v1 import Ecosystem
 from semgrep.semgrep_interfaces.semgrep_output_v1 import FoundDependency
 from semgrep.semgrep_interfaces.semgrep_output_v1 import Pypi
 
+# These use [until] instead of [upto] because [upto] only works on single characters
+# We don't care about the contents of any list or object values right now
 
-# A value in a key-value pair.
+# Using ]\n allows us to avoid issues with closing brackets inside strings
```

Comment:
Is ]\n in a string not allowed?

Reply:
```
[
   coverage[
      toml
   ]
]
```

Reply:
Well... I don't know for sure honestly, due to the lack of docs. It seems very unlikely to me that poetry would auto-generate an unquoted newline in a string, but I might be wrong. Perhaps better to just write a legit parser for the contents of these things...

Reply:
It would be
```
[
    "coverage[
        toml
    ]"
]
```
which seems very weird to me. Is there any format that allows unescape newlines in single quote strings?

Reply:
I'm gonna go with it for now